packages/wdio-sync/tests/index.test.js

```diff
@@ -1,127 +1,46 @@
-import sync, { executeSync, runSync } from '../src'
+import deasync from 'deasync'
 
-beforeAll(() => {
-    if (!global.browser) {
-        global.browser = {}
-    }
-})
-
-it('exports a function to make async tests sync', async () => {
-    const fakeFn = jest.fn().mockReturnValue('foobar')
-    expect(await sync(fakeFn)).toBe('foobar')
-})
-
-describe('executeSync', () => {
-    it('should pass with default values and regular fn', async () => {
-        global.browser._NOT_FIBER = true
-        expect(await executeSync.call({}, () => 1, {})).toEqual(1)
-        expect(global.browser._NOT_FIBER).toBe(undefined)
-    })
-
-    it('should pass with args and async fn', async () => {
-        const fn = async arg => arg
-        const scope = {}
-        expect(await executeSync.call(scope, fn, { limit: 1, attempts: 0 }, [2])).toEqual(2)
-        expect(scope.wdioRetries).toBe(0)
-    })
-
-    it('should pass without scope', async () => {
-        const fn = async arg => arg
-        expect(await executeSync(fn, { limit: 1, attempts: 0 }, [2])).toEqual(2)
-    })
+import waitForPromise from '../src'
 
-    it('should filter stack on failure', async () => {
-        global.browser._NOT_FIBER = true
-        let error
-        try {
-            await executeSync.call({}, () => { throw new Error('foobar') }, {})
-        } catch (err) {
-            error = err
-        }
-        expect(error.stack).not.toContain('at new Promise (<anonymous>)')
-        expect(global.browser._NOT_FIBER).toBe(undefined)
-    })
+jest.mock('deasync', () => ({
+    loopWhile: jest.fn().mockImplementation((fn) => fn())
```

Comment:
Shouldn't we clear the mock in afterEach hook?

Reply:
fixed